__tests__/integration/txLogger.js

```diff
@@ -0,0 +1,137 @@
+import * as Path from "path";
+
+const fsold = require('fs');
+const fs = fsold.promises;
+
+export const loggers = {
+  /**
+   * @type: TxLogger
+   */
+  test: null
+}
+
+/**
+ * A logger for every transaction on the integration tests for debugging.
+ */
+export class TxLogger {
+  #instanceFilename;
+  #fileFullPath;
+
+  get filename() {
+    return this.#instanceFilename;
+  }
+
+  /**
+   * Builds the log filename based on current time and an optional title
+   * @param {string} [title] Optional title. Keep it short and simple for readability
+   */
+  constructor(title) {
+    const date = new Date();
+    const timestamp = date.toISOString()
+      .replace(/-/g,'') // Remove date separator
+      .replace(/:/g,'') // Remove hour separator
+      .split('.')[0] // Get only the seconds integer
+    const additionalTitle = `-${title}` || ''
+    const filename = `${timestamp}${additionalTitle}-integrationTest.log`;
+    this.#instanceFilename = filename;
+  }
+
+  /**
+   * Initializes the log file on a specified folder
+   * @param {string} rootFolder
+   * @param {string} [testName] Optional title to include in the filename
+   * @returns {Promise<void>}
+   */
+  async init(rootFolder, testName) {
+    if (!rootFolder) throw new Error(`Root folder is mandatory`)
+
+    // Create the temporary files directory, if it does not exist
+    const tmpDir = Path.join(rootFolder, `/tmp/`);
+
+    // The promisified version of fs.assign is incompatible with our babel-node setup for some reason.
+    // Reverting back to using the old callback solution for compatibility
+    const dirPromise = new Promise((resolve) => {
+      try {
+        fsold.access(tmpDir, async (err, results) => {
+          if (err) {
+            fs.mkdir(tmpDir)
+              .catch(err2 => {
+                console.error(`Now that's an error: ${err2.stack}`);
+              });
+          }
+        })
+      } catch (err) {
+        console.error('Untreated error on fs.access: ', err.stack);
```

Comment:
The promise will be rejected here or should we add the reject?

Reply:
Since the logfiles are not mandatory for the tests, these errors were being ignored.
Added a proper rejection to the function.